skyve-war/src/main/java/modules/admin/User/UserBizlet.java

```diff
@@ -94,6 +94,19 @@ public void preRerender(String source, UserExtension bean, WebContext webContext
 			}
 		}
 
+		if (User.newPasswordPropertyName.equals(source)) {
+			String newPassword = bean.getNewPassword();
+			if (newPassword != null) {
+				Configuration c = Configuration.newInstance();
+				if (c.isCheckForBreachedPasswordsEnabled()) {
+					if (HIBPPasswordValidator.isPasswordPwned(newPassword)) {
+						webContext.growl(MessageSeverity.warn,
+								"WARNING: The password you have entered has been compromised in a data breach. For security, we recommend selecting a stronger password.");
```

Comment:
Because this message is used in a couple of places, I would probably externalise the string. Makes it easier to update the warning in one place (and translate).